static/js/contextMenu.js

```diff
@@ -0,0 +1,355 @@
+var selectedLineNumber;
+var lines = 0;
+var loginOrEmails = [];
+
+exports.aceSelectionChanged = function(hook, context){
+  var selStart = context.rep.selStart;
+  var selEnd = context.rep.selEnd;
+  if((selStart[0] !== selEnd[0]) || (selStart[1] !== selEnd[1])){
+    iT.show(selStart, selEnd);
+  }else{
+    iT.hide();
+  }
+}
+
+exports.postAceInit = function(hook, context) {
+  
+  attachContextMenu();
+  lines = 0;
+  $('iframe[name="ace_outer"]').contents().find('iframe').contents().find("#innerdocbody").contents().each(function() {
+    lines++;
+  });
+}
+
+function attachContextMenu() {
+  var count = 0;
+  var innerDocBody = $('iframe[name="ace_outer"]').contents().find('iframe').contents().find("#innerdocbody");
+  
+  $(innerDocBody).children().each(function() {
+    var currentLine = count;
+    $(this).off('contextmenu');
+    $(this).bind('contextmenu', function(e) {
+      e.preventDefault();
+      onRightClick(currentLine);
+      drawContextMenu(e.clientX, e.clientY);
+    })
+    count++;
+  });
+}
+
+jQuery.fn.selectText = function() {
+  var obj = this[0];
+  var selection = obj.ownerDocument.defaultView.getSelection();
+  var range = obj.ownerDocument.createRange();
+  range.selectNodeContents(obj);
+  selection.removeAllRanges();
+  selection.addRange(range);
+  return this;
+}
+
+jQuery.fn.copyText = function() {
+  var obj = this[0];
+  var ownerDocument = obj.ownerDocument;
+  var textarea = ownerDocument.querySelector('#text_to_copy');
+  textarea.focus({preventScroll: true});
+  textarea.select();
+  ownerDocument.execCommand('copy');
+}
+
+exports.aceSelectionChanged = function(hook, context){
+  var selStart = context.rep.selStart;
+  var selEnd = context.rep.selEnd;
+  if((selStart[0] !== selEnd[0]) || (selStart[1] !== selEnd[1])){
+    selectedLineNumber = selStart[0];
+  }else{
+    var padOuter = $('iframe[name="ace_outer"]').contents().find("body");
+    var contextMenu = $(padOuter).find("#context_menu");
+    selectedLineNumber = null;
+    contextMenu.hide();
+  }
+}
+
+exports.aceKeyEvent = function(hook, context){
+  if((context.evt.key == "Enter"||context.evt.key == "Backspace") && context.rep.alines.length != lines){
+    lines = context.rep.alines.length;
+    attachContextMenu();
+  }
+}
+
+function onRightClick(lineNumber) {
+  if (selectedLineNumber) {
+    return;
+  }
+  var innerDocBody = $('iframe[name="ace_outer"]').contents().find('iframe').contents().find("#innerdocbody");
+  $(innerDocBody).children().eq(lineNumber).selectText();
+  selectedLineNumber = lineNumber;
+}
+
+
+function drawNotifyModal() {
+  var padOuter = $('iframe[name="ace_outer"]').contents().find("body");
+  var notifyModal = $(padOuter).find('#notifyModal');
+  if (notifyModal.length === 0) {
+    var baseURL = window.location.href.slice(0, window.location.href.split('/', 3).join('/').length);
+    padOuter.append('<div id="notifyModal" class="modal"></div>');
+    notifyModal = $(padOuter).find('#notifyModal');
```

Comment:
This part seems a bit hard to maintain.  Is it possible to just prepare the entire modal html text using a JavaScript template, and then use set innerHTML for a div?  I suggest the modal itself might be better off implemented as a separate class from contextMenu.